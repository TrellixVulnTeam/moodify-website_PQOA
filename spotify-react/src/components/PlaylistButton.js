import React from "react";

export default function PlaylistButton(props) {

    return (
        <div className="playlist-button" data-testid="playlist-button">
            {props.token ?
                <div className={""}>
                    <h5 className={"text-2xl text-indigo-400 pt-3 pb-8"}>Click to make a playlist</h5>
                    <form onSubmit={props.testingFunction}>
                        <select className={"shadow appearance-none border rounded-l-lg w-auto py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-xl bg-transparent"} name={"emotion"} id={"emotion-playlist"} onClick={e => props.setPlaylistEmotion(e.target.value)} data-testid="select-menu">
                            <option className={""} disabled>Choose an Emotion</option>
                            <option className={""} value="Anger" data-testid="option">Anger</option>
                            <option className={""} value="Fear" data-testid="option">Fear</option>
                            <option className={""} value="Joy" data-testid="option">Joy</option>
                            <option className={""} value="Sadness" data-testid="option">Sadness</option>
                        </select>
                        <button
                            className={"shadow bg-indigo-500 border rounded-r-lg hover:bg-indigo-700 focus:shadow-outline leading-tight focus:outline-none text-white py-2 px-3 text-xl"}
                            type={"submit"}>Make
                        </button>
                    </form>
                </div>
                :
                ""
            }
        </div>
    )
}
