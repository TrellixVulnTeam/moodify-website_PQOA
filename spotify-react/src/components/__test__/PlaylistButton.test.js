import React from "react";
import ReactDOM from 'react-dom'
import PlaylistButton from '../PlaylistButton'

import {cleanup, render, fireEvent, getByTestId, getAllByTestId} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";


it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<PlaylistButton/>, div)
})

afterEach(cleanup);

it("renders playlist-button correctly", () => {
    const {getByTestId} = render(<PlaylistButton/>)
    expect(getByTestId("playlist-button")).toBeDefined()
})

it("matches snapshot", () =>{
    const tree = renderer.create(<PlaylistButton token={"testing token"}/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("choosing option", () =>{
    const {getByTestId, getAllByTestId} = render(<PlaylistButton token={"testing token"}/>)
    fireEvent.change(getByTestId('select-menu'), {target: { value: "Anger"}})
    let options = getAllByTestId('option')
    expect(options[0].selected).toBeTruthy()
    expect(options[1].selected).toBeFalsy()
    expect(options[2].selected).toBeFalsy()
    expect(options[3].selected).toBeFalsy()
})

