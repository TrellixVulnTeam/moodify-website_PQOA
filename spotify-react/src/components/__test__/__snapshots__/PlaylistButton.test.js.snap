// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="playlist-button"
  data-testid="playlist-button"
>
  <div
    className=""
  >
    <h5
      className="text-2xl text-indigo-400 pt-3 pb-8"
    >
      Click to make a playlist
    </h5>
    <form>
      <select
        className="shadow appearance-none border rounded-l-lg w-auto py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-xl bg-transparent"
        data-testid="select-menu"
        id="emotion-playlist"
        name="emotion"
        onClick={[Function]}
      >
        <option
          className=""
          disabled={true}
        >
          Choose an Emotion
        </option>
        <option
          className=""
          data-testid="option"
          value="Anger"
        >
          Anger
        </option>
        <option
          className=""
          data-testid="option"
          value="Fear"
        >
          Fear
        </option>
        <option
          className=""
          data-testid="option"
          value="Joy"
        >
          Joy
        </option>
        <option
          className=""
          data-testid="option"
          value="Sadness"
        >
          Sadness
        </option>
      </select>
      <button
        className="shadow bg-indigo-500 border rounded-r-lg hover:bg-indigo-700 focus:shadow-outline leading-tight focus:outline-none text-white py-2 px-3 text-xl"
        type="submit"
      >
        Make
      </button>
    </form>
  </div>
</div>
`;
